sbrnet_core/sbrnet/noisemodel.py

```diff
@@ -0,0 +1,49 @@
+from typing import Tuple
+import torch
+import torch.nn as nn
+from torch import Tensor, Module
+
+
+class PoissonGaussianNoiseModel(Module):
+    """Poisson-Gaussian noise model for CM2 sensor: https://ieeexplore.ieee.org/document/4623175"""
+
+    # from calibrated CM2 SONY imx-226 sensor on [0,1] normalized data
+    A_STD = 5.7092e-5
+    A_MEAN = 1.49e-4
+    B_STD = 2.7754e-6
+    B_MEAN = 5.41e-6
+
+    def __init__(self, a_std=A_STD, a_mean=A_MEAN, b_std=B_STD, b_mean=B_MEAN):
+        super().__init__()
+
+        self.a_std = a_std
+        self.a_mean = a_mean
+        self.b_std = b_std
+        self.b_mean = b_mean
+
+    def forward(self, stack: Tensor, rfv: Tensor) -> Tuple[Tensor, Tensor]:
+        """forward(x) = x + sqrt(a*x + b) * N(0,1), where a and b are calibrated parameters
+
+        Args:
+            stack (Tensor): the noise-free stack of light field views normalized to [0,1]
+            rfv (Tensor): the noise-free refocused volume normalized to [0,1]
+
+            note: ideally we would perform the backprojection on the raw LF views with the noise added
+            to attain the RFV. we simplify this by just adding the same noise to both, and dividing the std by the
+            sqrt of the number of views being added to form the RFV, like an "denoising" approach.
+        Returns:
+            Tuple[Tensor, Tensor]: stack and rfv with poisson-gaussian noise added
+        """
+        a = torch.randn(1) * self.a_std + self.a_mean
```

Comment:
This could be problematic when training since $\forall x \in$ { $y | ay \le -b$ } then $ax + b \le 0$ then it's no longer stochastic. There should be a better way of modeling the uncertainty of the camera statistics such that don't allow for the process to become deterministic.

Reply:
i had a and b random because when i calibrate the a and b values, i use multiple measurements from the sensor, and each measurement has different a and b values, so its more modeling the "measurement uncertainty" for these parameters. 

im not sure what you mean with the math

Reply:
Pretty much what I'm saying is that modeling `a` and `b` as gaussian is what causes the numerical stability issue. And doing `clamp` on them causes no noise to be added when you get a negative value (which implies non-sensical negative variance). The problem of having `a` and `b` as independent gaussian means $a \cdot x + b$ is also gaussian. And since gaussians can be negative or positive valued you cannot take the square root of the process and using clamp doesn't have a clear theoretical intuition since it ignores the random process even though it addresses the negative square root. It is a more general comment that is outside the scope of this PR, but a comment on the theoretical formulation of the model.

Reply:
i see. i think then it's probably fine to just use the mean of the calibrated parameters then